content/learn/beginner/control-flow-1.md

```diff
@@ -0,0 +1,213 @@
++++
+title = "Control Flow 1"
+weight = 230
++++
+
+Until now, we have just written some linear code that executes from top to bottom. However, in real-world applications,
+we often need to make decisions based on certain conditions. This is where *control flow* comes in.
+
+You already have an intuitive understanding of control flow. For instance, when you cross the street, you look left and
+right to see if any cars are coming. If there are no cars, you cross the street. If there are cars, you wait until they
+have passed, then check again. This is a simple example of a decision-making process.
+
+## Comparison Actions
+
+In programming, we can make decisions by comparing values to each other. We can compare numbers, strings, and other data
+types. The result of a comparison is a *boolean* value, which is either `true` or `false`.
+
+{{< callout context="caution" title="Comparing across Types" icon="outline/alert-triangle" >}}
+
+Just like you cannot quite compare apples and oranges, you cannot compare values of different types. For instance, you
+cannot compare a number to a string. The bot will throw an error if you try to do so, you will have to convert either of
+them to the other type first.
+
+{{< /callout >}}
+
+We provide the following comparison functions in Custom Commands:
+
+- `eq` (equals `==`)
+- `ne` (not equals `!=`)
+- `lt` (less than `<`)
+- `le` (less than or equal to `<=`)
+- `gt` (greater than `>`)
+- `ge` (greater than or equal to `>=`)
+
+These functions can only compare basic data types as introduced in [Data Types 1](/learn/beginner/datatypes-1). Strings
+are compared using the [underlying Unicode values](https://en.wikipedia.org/wiki/List_of_Unicode_characters), stepping
+through the strings byte by byte. Please refer to the [functions documentation](/docs/reference/templates/functions)
+for the syntax of these functions.
+
+### If Statements
```

Comment:
I would rather put this section first, noting that `eq` is a comparison function that checks whether its arguments are equal and that the full list of comparison functions will be enumerated later.